### DISCUSSION TITLE
LastItemRect.Min and Max are not accessible anymore in 1.84

### DISCUSSION THREAD

**Discussion by SadE54** [ocornut/imgui#4526] [New Users Build/Link/Run issues ONLY!]
I guess it's only a API changes but how to access it now from last versions ?

**Comment by PathogenDavid** [comment]
Edit: As Omar said, you should really be using `GetItemRectMin`/`GetItemRectMax` instead. (I meant to mention originally this but got tunnel vision.)

------------

Keep in mind that `imgui_internal.h` is internal to Dear ImGui and does not have a stable API:

https://github.com/ocornut/imgui/blob/17a7084b57f9713ec6538881079d6c5a1b6b3598/imgui_internal.h#L4

`LastItemRect` was moved out of the per-window temporary data and into the global context in https://github.com/ocornut/imgui/commit/6b8a059fc99f27438010c92aae0b05903f27b7d1.

So before where you might use `GetCurrentWindow()->DC.LastItemRect` you now need to use `GetCurrentContext()->LastItemData.Rect`.

**Comment by ocornut** [comment]
Note that this has always been exposed as GetItemRectMin(), GetItemRectMax().

**Reply by SadE54** [reply]
Thanks for the answers !
btw , it's the way it's implemented in https://github.com/dfranx/ImFileDialog that I'm trying to get it working :)

**Reply by SadE54** [reply]
The current code in the concerned library is 
```
ImGuiContext& g = *GImGui;
ImGuiWindow* window = g.CurrentWindow;
...
 window->DrawList->AddRectFilled(window->DC.LastItemData.Rect.Min, window->DC.LastItemRect.Max, ImGui::ColorConvertFloat4ToU32(ImGui::GetStyle().Colors[active ? ImGuiCol_HeaderActive : ImGuiCol_HeaderHovered]));
```

I'm not sure about the good change in this case 🤔
